packages/docs/src/pages/guides/jsx-pragma.mdx

```diff
@@ -55,3 +64,120 @@ The preferred way of using these functions is by adding the custom pragma commen
 See the [`sx` prop][] docs to learn more.
 
 [`sx` prop]: /sx-prop
+
+## Automatic JSX runtime
+
+React v17 introduced a
+[new JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html),
+also called "automatic runtime" (backported to React v16.14.0) If you use any of
+those versions together with a configurable transpiler (such as Babel or
+TypeScript), you can configure JSX to use the automatic runtime (globally, for
+your entire app), and no longer need to use the custom pragma comments in your
+files to use `sx`.
+
+Here are a few examples showing how to do that depending how you build your
+application:
+
+### Using @babel/preset-react
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/preset-react',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-preset-react#react-automatic-runtime)
+
+### Using @babel/plugin-transform-react-jsx
+
+This plugin is included in the preset above, but can also used as a standalone
+babel plugin.
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/plugin-transform-react-jsx',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#react-automatic-runtime)
+
+### Using TypeScript
+
+If you use TypeScript to transpile your source code with `tsc` (or only
+typecheck), or for instance to run tests with `ts-jest`
```

Comment:
Sorry mate, I was under impression we already merged this PR.

Could we write something like "If you use TypeScript to transpile your JSX"?
A lot of people are using TypeScript together with Babel, and they don't need to configure jsxImportSource globally twice.

Reply:
You still need it for instance for eslint and other tools like `ts-jest` that do not use TS, even if you use `@babel/preset-typescript` to build, right?

Reply:
Good point. I assumed people would use `babel-jest`, but `typescript-eslint` will probably need it in the tsconfig when it's doing typechecking.

Reply:
I think it's okay now. @lachlanjc could you take a look at this PR when you have a moment?